### PR TITLE
Add replacements for tag deletions.

### PR DESCRIPTION
**Pull request by bretthoerner**

### CODE REVIEW

File: snuba/replacer.py [hunk 1]
```diff
@@ -277,3 +280,63 @@ def process_unmerge(message):
...
+            )
+        else:
+            select_columns.append(col.escaped)
+
+    query_args = {
+        'all_columns': ', '.join(ALL_COLUMN_NAMES),
+        'select_columns': ', '.join(select_columns),
+        'project_id': message['project_id'],
+        'tag_str': escape_string(tag),
+        'tag_column': escape_col(tag),
```
Comment (alex-hofsteede) [inline snuba/replacer.py]: I think we maybe need to do a little more translation here? Probably have to at least lookup the tag in `TAG_COLUMN_MAP` to get the clickhouse column name from the tag provided.

A test to make sure that stuff like `browser.name` / `browser_name` works would be good too.

**Review by alex-hofsteede — CHANGES_REQUESTED** [review]
One small thing about the tag name translations, and a question about the scope of the deletion

File: snuba/replacer.py [hunk 2]
```diff
@@ -277,3 +280,63 @@ def process_unmerge(message):
...
+    if not tag:
+        return None
+
+    assert isinstance(tag, six.string_types)
+    timestamp = datetime.strptime(message['datetime'], settings.PAYLOAD_DATETIME_FORMAT)
+    is_promoted = tag in PROMOTED_TAGS['tags']
+
+    where = """\
+        WHERE project_id = %(project_id)s
+        AND received <= CAST('%(timestamp)s' AS DateTime)
```
[the remark this answers is not in the document]
Comment (bretthoerner) [reply snuba/replacer.py]: It's project wide:

https://github.com/getsentry/sentry/blob/e065f263e7c51941f7e641bb71a1e8ab3d58322b/src/sentry/tagstore/v2/backend.py#L594

Trash can buttons here:

https://sentry.io/settings/sentry/projects/sentry/tags/

[the remark this answers is not in the document]
Comment (bretthoerner) [reply snuba/replacer.py]: Which, btw, was the worry about "this one button deletes a tag we used on every single event in our 100,000,000 event project" in regards to scrubbing nodestore.

File: snuba/replacer.py (hunk 1 above)
Comment (bretthoerner) [reply snuba/replacer.py]: Good catch, fixed.

**Review by alex-hofsteede — APPROVED** [review]
LGTM

**Pull request merged**